src/Microsoft.Azure.WebJobs.Host/Scale/TriggerMetadataProvider.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Host.Scale
+{
+    internal class TriggerMetadataProvider : ITriggerMetadataProvider
+    {
+        private readonly IEnumerable<TriggerMetadata> _triggerMetadata;
+
+        public TriggerMetadataProvider(IEnumerable<TriggerMetadata> triggerMetadata)
+        {
+            _triggerMetadata = triggerMetadata;
+        }
+
+        public IEnumerable<TriggerMetadata> GetTriggerMetadata()
```

Comment:
Do we want to filer by trigger type?

Reply:
I don't think so. An implementor of this interface has a single job - return all the trigger metadata. That metadata may be used in a variety of ways as needed by consumers. If needed, we could add some extension methods to simplify common operations. Once we have some extensions on the new APIs we can look and decide how we can simplify their logic I think.

Reply:
Discussed offline, let's keep the interface as simple as we can